pytest_mozwebqa/cloud/saucelabs.py

```diff
@@ -0,0 +1,130 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from ConfigParser import ConfigParser
+import os
+import json
+
+from py.xml import html
+import requests
+from selenium import webdriver
+
+
+class SauceLabs(object):
+
+    name = 'Sauce Labs'
+
+    def __init__(self):
```

Comment:
I feel like some of the code in here could potentially be refactored into a parent class, as a lot of the same things are being done for both `BrowserStack` and `SauceLabs`.   

Examples include:
- Both read from `setup.cfg` and check for a `username` and key. The key does have a different name for each provider, but you could normalize those if you wished.
- Both check environment variables for the same username and key, and use the same logic for choosing what to use for the username and key, as well as raising an exception if either is missing.

Having said that, the resulting code would probably be less readable than what you have here, as one would have to look at both objects at the same time to figure out exactly what is happening, so I am not unhappy with what you've done here if you'd like to keep the (minor) duplication.